mcu/src/dac.c

```diff
@@ -0,0 +1,75 @@
+#include <msp430.h>
+
+#include "dac.h"
+
+#include <stdio.h>
+
+void init_dac(struct dac_t dac)
+{
+    // I used section 2.5 of the following app note to figure out
+    // how to configure the "smart analog combo" to operate in DAC mode
+    // https://www.ti.com/lit/an/slaa833a/slaa833a.pdf
+
+    // Define the register base addresses for the particular SACx module
+    // that the dac struct is set up for. The base_addr in the dac struct
+    // defines the base address of the SAC module. The register offsets for
+    // each module are the same, so we use the SAC0 offset defintions, which
+    // are defined in msp420fr2355.h
+    uint16_t *DAC_REG = (uint16_t *) (dac.sac_base_addr + OFS_SAC0DAC);
+    uint16_t *OA_REG = (uint16_t *) (dac.sac_base_addr + OFS_SAC0OA);
+    uint16_t *PGA_REG = (uint16_t *) (dac.sac_base_addr + OFS_SAC0PGA);
+
+    // Define the base addresses for the port multiplexing registers, PxSEL1 and
+    // PxSEL0. These offsets are the same for all ports, and are defined in
+    // msp430fr2355.h. We use the port 1 offsets.
+    uint16_t *PX_SEL0 = (uint16_t *) (dac.port_base_addr + OFS_P1SEL0); 
+    uint16_t *PX_SEL1 = (uint16_t *) (dac.port_base_addr + OFS_P1SEL1); 
+
+    // Enable the 1.5 V internal reference
+    /* NOTE: the MSP430FR2355 internal reference defaults to 1.5 V, but
+             but can be changed to 2.0 or 2.5V. */
+    // This will be enabled each time a DAC is initialized, but that's okay
+    // because the bit will always be a 1 after the first initialization.
+    PMMCTL2 |= INTREFEN;
```

Comment:
We should add a DEFINE constant for the 1.5 V reference. The main purpose of this would be for computing the scaling constant used in `set_dac_voltage()`